jobs/fxci-taskcluster-export/fxci_etl/loaders/bigquery.py

```diff
@@ -71,12 +136,16 @@ def insert(self, records: list[Record] | Record):
         try:
             # Load previously failed records from storage, maybe the issue is fixed.
             for obj in json.loads(self._record_backup.download_as_string()):
-                records.append(Record.from_dict(obj))
+                table = obj.pop("table")
+                records.append(Record.from_dict(table, obj))
         except NotFound:
             pass
 
-        tables = defaultdict(list)
+        tables = {}
         for record in records:
+            if record.table not in tables:
+                self.ensure_table(record.table, record.__class__)
```

Comment:
How many rows are there per run usually?
Running this per row seems suboptimal. Could we do this once per each table that we have in the set of rows?

Reply:
It's not super clear, but this is running one per table because on the next iteration `record.table` will be in `tables` and we won't enter this `if` block.

Reply:
I agree it would be nice to just do this ensuring once at startup or something. But the `BigQueryLoader` class is designed to not know about any specific tables and I was trying to avoid hardcoding them.